runtime/compiler/optimizer/J9TransformUtil.cpp

```diff
@@ -1019,7 +1021,9 @@ J9::TransformUtil::canFoldStaticFinalField(
 
       // General aggressive folding is disabled by environment variable. Still
       // try to apply it to static final fields with declared type VarHandle,
-      // as before.
+      // as before. Additionally for JDK21+, fold final fields related to
+      // VarHandleSegmentView operations, which include ValueLayout, ValueLayouts
+      // and MemorySegment.
```

Comment:
Checking for these types is fine, but it doesn't seem like it should be necessary, since this part only runs when `TR_disableAggressiveSFFF17` is set anyway